docs/next/guides/upgrade-and-migration/migrating-from-11.1-to-12.0.md

```diff
@@ -12,3 +12,148 @@ pageClass: migration-guide
+
+// update your other options separately
+hot.updateSettings({
+  width: 400,
+  height: 300,
+});
+```
+
+## Step 2: `updatePlugin()`
+
+Handsontable [12.0.0](https://github.com/handsontable/handsontable/releases/tag/12.0.0) changes the way the [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method reacts to [`updateSettings()`](@/api/core.md#updatesettings) calls.
+
+#### Before
+
+Every [`updateSettings()`](@/api/core.md#updatesettings) call (even with an empty object passed as new settings) triggers 
+the [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method for each enabled plugin.
+
+As a result, whenever you call [`updateSettings()`](@/api/core.md#updatesettings), all enabled plugins get updated.
+
+#### After
+
+A plugin gets updated only if you update settings related to that particular plugin.
+
+Only the following plugins still get updated on every [`updateSettings()`](@/api/core.md#updatesettings) call:
+  - [`AutoColumnSize`](@/api/autocolumnsize.md)
+  - [`AutoRowSize`](@/api/autorowsize.md)
+  - `TouchScroll`
+  - [`UndoRedo`](@/api/undoredo.md)
+
+#### How to upgrade
+
+If you want a plugin (e.g. [`ContextMenu`](@/api/contextmenu.md)) to still get updated every time you call [`updateSettings()`](@/api/core.md#updatesettings), use the [`afterUpdateSettings`](@/api/hooks.md#afterupdatesettings) hook to call your plugin's [`updatePlugin()`](@/api/autocolumnsize.md#updateplugin) method:
+
+```js
+afterUpdateSettings() {
+  contextMenu.updatePlugin();
+}
+```
+
+## Step 3: `afterDocumentKeyDown`
+
+Handsontable [12.0.0](https://github.com/handsontable/handsontable/releases/tag/12.0.0) changes the way the [`afterDocumentKeyDown`](@/api/hooks.md#afterdocumentkeydown) hook works.
+
+#### Before
+
+When you use a keyboard to navigate an internal Handsontable instance (e.g. a [context menu](@/guides/accessories-and-menus/context-menu.md)), the [`afterDocumentKeyDown`](@/api/hooks.md#afterdocumentkeydown) hook gets fired on every key press.
+
+```js
+afterDocumentKeyDown() {
+  // when you navigate the grid, the console message gets logged on every key press
+  // when you navigate the context menu, the console message also gets logged on every key press
+  console.log('key pressed');
+}
+```
+
+#### After
+
+When you use a keyboard to navigate an internal Handsontable instance (e.g. a [context menu](@/guides/accessories-and-menus/context-menu.md)), the [`afterDocumentKeyDown`](@/api/hooks.md#afterdocumentkeydown) hook may not get fired anymore.
+
+```js
+afterDocumentKeyDown() {
+  // when you navigate the grid, the console message gets logged on every key press
+  // when you navigate the context menu, the console message doesn't get logged at all
+  console.log('key pressed');
+}
+```
+
+This change may affect navigating the following areas of Handsontable:
+- [Context menu](@/api/contextmenu.md)
+- [Dropdown menu](@/api/dropdownmenu.md)
+- [Handsontable editor](@/guides/cell-types/handsontable-cell-type.md)
```

Comment:
We handle editor's shortcuts in the context of "main Handsontable", so this problem doesn't occur for `Handsontable editor`, `DropdownEditor`, `AutocompleteEditor` (all of them use HOT internally).

Reply:
@wszymanski - how about this: [ae9d4ed](https://github.com/handsontable/handsontable/pull/9271/commits/ae9d4ed96d98583bfb5950525eaaa0422d4f437d)?